bika/lims/browser/stickers.py

```diff
@@ -168,8 +171,22 @@ def getAvailableTemplates(self):
                  'title': <teamplate_title>,
                  'selected: True/False'}
         """
-        seltemplate = self.getSelectedTemplate()
+        # Getting adapters for current context. those adapters will return
+        # the desired sticker templates for the current context:
+        try:
+            adapters = getAdapters((self.context, ), IGetStickerTemplates)
+        except ComponentLookupError:
+            logger.info('No IGetStickerTemplates adapters found.')
+            adapters = None
         templates = []
+        if adapters is not None:
+            # Gather all templates
+            for name, adapter in adapters:
+                templates += adapter(self.request)
+        if templates:
+            return templates
```

Comment:
As I said in a previous comment, I can't return until the end of the method.

`adapters` variable can be a generator or `None`. So, if `getAdapters` doesn't fail, `adapters` variable will be an object and consecuently it will always be true.